FWCore/GuiBrowsers/python/Vispa/Gui/ConnectableWidget.py

```diff
@@ -289,15 +289,15 @@ def sinkPorts(self):
         return [port for port in self._ports if port.portType() == "sink"]
         def isSink(port):
             return port.portType() == 'sink'
-        return filter(isSink, self._ports)
+        return list(filter(isSink, self._ports))
```

Comment:
Do we understand this change? Does `filter` return a generator in python 3 while the old `filter` returned a list?

Reply:
so now I understand you consider this a problem. not sure what alternatives there are. I'll have a look

Reply:
seems an easy replacement is 

return [item for item in self._ports if isSink(item)]

I'd propose to handle the 13 changes (mostly in obsolete code and all in non-performance critical code) for a followup pr.

Reply:
I concur.